compiler/noirc_evaluator/src/ssa/opt/basic_conditional.rs

```diff
@@ -0,0 +1,522 @@
+use std::collections::{HashSet, VecDeque};
+
+use acvm::AcirField;
+use fxhash::FxHashMap as HashMap;
+use iter_extended::vecmap;
+
+use crate::ssa::{
+    ir::{
+        basic_block::BasicBlockId,
+        cfg::ControlFlowGraph,
+        dfg::DataFlowGraph,
+        function::{Function, FunctionId},
+        function_inserter::FunctionInserter,
+        instruction::{BinaryOp, Instruction, TerminatorInstruction},
+        post_order::PostOrder,
+        value::ValueId,
+    },
+    Ssa,
+};
+
+use super::flatten_cfg::Context;
+#[derive(Debug, Clone)]
+struct BasicConditional {
+    block_entry: BasicBlockId,
+    block_then: Option<BasicBlockId>,
+    block_else: Option<BasicBlockId>,
+    block_exit: BasicBlockId,
+}
+
+impl Ssa {
+    #[tracing::instrument(level = "trace", skip(self))]
+    pub(crate) fn flatten_basic_conditionals(mut self) -> Ssa {
```

Comment:
It would be helpful to add a doctoring with a basic example of what this pass achieves. For example when we walk someone through the passes, it's a quick way to explain what we mean without having to look up unit tests. Maybe examples are not realistic in more complex cases, and having them here but not there is just irregular, but at sentence or two of what it's looking for and what tradeoffs it's considering would be great IMO.

Reply:
Done